src/resmom/mom_comm.c

```diff
@@ -5409,7 +5409,7 @@ im_request(int stream, int version)
 			resc_idx = -1;
 			for (i=0; i < pjob->ji_numrescs; i++) {
 				if ((pjob->ji_resources[i].nodehost != NULL) &&
-				    (compare_short_hostname(pjob->ji_resources[i].nodehost, nodehost) == 0)) {
+				    (strcmp(pjob->ji_resources[i].nodehost, nodehost) == 0)) {
```

Comment:
Would it be better to enhance compare_short_hostname() to accommodate IP addresses as well? Then it would work in other cases where it's called. If not, we should use strncmp() here.

Reply:
@mike0042 : yes, I agree, that compare_short_hostname() should recognize if IP addresses are involved and if so, just do the strcasecmp(). I've updated the fix and pushed a new commit. Please review.